scalyr_agent/builtin_monitors/journald_utils.py

```diff
@@ -82,47 +85,101 @@ def create_config_matcher(self, conf):
         in data, None otherwise
         """
         config = copy.deepcopy(conf)
-        if "journald_unit" not in config:
-            config["journald_unit"] = ".*"
+
+        journald_unit = config.get("journald_unit", None, none_if_missing=True)
+        journald_globs = config.get("journald_globs", None, none_if_missing=True)
+
+        # User shouldn't be able to specify both `journald_unit` and
+        # `journald_globs`
+        if journald_unit is not None and journald_globs is not None:
+            raise BadConfiguration(
+                "Cannot specify both journald_unit and journald_globs",
+                "journald_unit",
+                "invalidConfigError",
+            )
+
+        # If neither is specified, default to journald_unit matching
+        # everything
+        if journald_unit is None and journald_globs is None:
+            journald_unit = ".*"
+            config["journald_unit"] = journald_unit
+
+        match_hash = "monitor"
+        regex = None
+        if journald_unit is not None:
+            match_hash = six.text_type(hash(journald_unit))
+            if journald_unit == ".*":
```

Comment:
This is just thinking out loud and probably micro optimization we don't need (since it's an existing behavior), but we could also return a special version of the matcher  function which just always return ``True`` in case ``journald_unit == ".*"`` to avoid regex match operation (Which I believe should be very fast anyway due to the nature of the regex itself. I would assume Python C regex code already has some special case to bail out early in such scenarios, but not 100% especially since there are some edge case with new line matching where even if ``.*`` is used, whole input string would still need to be checked).

Reply:
Yep, that's an easy thing to do.